src/Lykke.Service.IroncladDecorator/OpenIdConnect/OpenIdValidators.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using IdentityModel;
+using IdentityModel.Client;
+using Lykke.Service.IroncladDecorator.Sessions;
+using Lykke.Service.IroncladDecorator.Settings;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Lykke.Service.IroncladDecorator.OpenIdConnect
+{
+    public class OpenIdValidators : IOpenIdValidators
+    {
+        private readonly ValidationSettings _validationSettings;
+        private readonly IDiscoveryCache _discoveryCache;
+        private readonly JwtSecurityTokenHandler _jwtSecurityTokenHandler;
+        private readonly OpenIdConnectProtocolValidator _connectProtocolValidator;
+
+        public OpenIdValidators(
+            ValidationSettings validationSettings,
+            IDiscoveryCache discoveryCache)
+        {
+            _validationSettings = validationSettings;
+            _discoveryCache = discoveryCache;
+            _connectProtocolValidator = new OpenIdConnectProtocolValidator
+            {
+                RequireTimeStampInNonce = false
+            };
+            _jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+            _jwtSecurityTokenHandler.InboundClaimTypeMap.Clear();
+        }
+
+        public async Task<IdTokenValidationResult> ValidateWebClientTokenResponseAsync(
+            UserSession userSession,
+            TokenResponse tokenResponse)
```

Comment:
Maybe this can be placed inside of TokenResponse model?

Reply:
Fixed